internal/ingress/annotations/annotations.go

```diff
@@ -49,39 +57,42 @@ const (
 	DefaultIngressClass = "kong"
 )
 
-func validIngress(ingressAnnotationValue, ingressClass string, skipClasslessIngress bool) bool {
-	// we have 2 valid combinations
-	// 1 - ingress with default class | blank annotation on ingress
-	// 2 - ingress with specific class | same annotation on ingress
-	//
-	// and 2 invalid combinations
-	// 3 - ingress with default class | fixed annotation on ingress
-	// 4 - ingress with specific class | different annotation on ingress
-	if ingressAnnotationValue == "" && !skipClasslessIngress && ingressClass == DefaultIngressClass {
+func validIngress(ingressAnnotationValue, ingressClass string, handling ClassMatching) bool {
+	switch handling {
+	case IgnoreClassMatch:
+		// class is not considered at all. any value, even a mismatch, is valid
 		return true
+	case ExactOrEmptyClassMatch:
+		// aka lazy. exact match desired, but empty permitted
+		return ingressAnnotationValue == "" || ingressAnnotationValue == ingressClass
+	case ExactClassMatch:
+		// what it says on the tin
+		// this may be another place we want to return a warning, since an empty-class resource will never be valid
+		return ingressAnnotationValue == ingressClass
+	default:
+		panic("invalid ingress class handling option received")
```

Comment:
```suggestion
		panic("invalid ingress class handling option received: " + handing)
```

Reply:
Can't, it's an int effectively. Leaving as-is rather than formatting--it won't really provide much info since it's, well, an int, so all we really care about is that it's not one of the ints we want. In the event we do hit the panic we mostly just care about the traceback, since whatever path passed the bad value is of more interest than the value given.